File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/entity/Cliente.java
package br.edu.utfpr.serverpedidosjava.entity;

import jakarta.persistence.*;

@Entity
public class Cliente {
    @Id
    @GeneratedValue
    private Integer id;
    private String nome;
    private String cpf;
    private String telefone;
    @OneToOne(mappedBy = "cliente", cascade = CascadeType.ALL)
    @PrimaryKeyJoinColumn
    private Endereco endereco;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public Endereco getEndereco() {
        return endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/entity/Cliente.kt
package br.edu.utfpr.serverpedidos.entity

import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.OneToOne
import jakarta.persistence.PrimaryKeyJoinColumn

@Entity
class Cliente(
    @Id
    @GeneratedValue
    var id: Int = 0,
    var nome: String = "",
    var cpf: String = "",
    var telefone: String = "",
    @OneToOne(mappedBy = "cliente", cascade = [CascadeType.ALL])
    @PrimaryKeyJoinColumn
    var endereco: Endereco? = null
)
